src/server/json_family.cc

```diff
@@ -259,8 +279,108 @@ OpResult<vector<OptBool>> OpToggle(const OpArgs& op_args, string_view key, strin
   return vec;
 }
 
+template <typename Op>
+OpResult<string> OpDoubleArithmetic(const OpArgs& op_args, string_view key, string_view path,
+                                    double num, Op arithmetic_op) {
+  OpResult<json> result = GetJson(op_args, key);
+  if (!result) {
+    return result.status();
+  }
+
+  bool is_result_overflow = false;
+  double int_part;
+  bool has_fractional_part = (modf(num, &int_part) != 0);
+  json output(json_array_arg);
+
+  auto cb = [&](const string& path, json& val) {
+    if (val.is_number()) {
+      double result = arithmetic_op(val.as<double>(), num);
+      if (isinf(result)) {
+        is_result_overflow = true;
+      }
+
+      if (val.is_double() || has_fractional_part) {
```

Comment:
why do you still need this `has_fractional_part` and `is_double` checks? we output doubles anyway. Where does it affect the logic?

Reply:
Actually, we output a string generated by the JSON library.  I don't want to convert the inner JSON item type to double if it's not nessecery.